providers/pinecone/src/airflow/providers/pinecone/hooks/pinecone.py

```diff
@@ -203,7 +203,7 @@ def get_pod_spec_obj(
             replicas=replicas,
             shards=shards,
             pods=pods,
-            pod_type=pod_type,
+            pod_type=pod_type,  # type: ignore[arg-type]
```

Comment:
Does it solve the issue or just hide the error?

Reply:
Solves it because we do not pass the other PodType object in our codebase, its always string that we pass.

Reply:
@eladkal do you have any comments on this?

Reply:
If PodSpec only expects PodType, it’s probably best to coerce on our end

```python
pod_type=pod_type if isinstance(pod_type, PodType) else PodType(pod_type)
```

This would throw a relatively readable error message if the user accidentally passes in an invalid value. It also guards against possible implementation detail change in pinecone if passing in a plain str no longer works.

Reply:
Yeah that seems to be a good idea.